### PR TITLE
add sys_arch_reboot() support for ARM64

### PR DESCRIPTION
**Pull request by JiafeiPan**
This PR is to add `sys_arch_reboot(int type) `support for ARM64.

If PSCI is enabled, it will leverage psci reset API to achieve system reboot, otherwise a weak dump reboot API is provided, and platform  can override these APIs if platform specified implementation provided.

Added imx93 board support for sample "`shell_module`" with PSCI enabled, use the shell kernel command can verify the reboot API sys_reboot():
`uart:~$ kernel reboot cold`

### CODE REVIEW

File: arch/arm64/core/reboot.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#ifdef CONFIG_PM_CPU_OPS_PSCI
+void __weak sys_arch_reboot(int type)
+{
+	unsigned char reset_type;
+
+	if (type == SYS_REBOOT_COLD)
+		reset_type = SYS_COLD_RESET;
+	else if (type == SYS_REBOOT_WARM)
+		reset_type = SYS_WARM_RESET;
+	else {
```
Comment (pdgendt) [inline arch/arm64/core/reboot.c:24]: As `clang-format` suggests, add curly braces here.

**Comment by JiafeiPan** [comment]
Dear Maintainers, would you please review this PR? thanks.

**Review by pdgendt — APPROVED** [review]

**Comment by JiafeiPan** [comment]
This PR is to replace implementation in https://github.com/zephyrproject-rtos/zephyr/pull/77056

**Review by dleach02 — APPROVED** [review]

**Comment by dleach02** [comment]
@carlocaione this PR needs your approval so we can merge

**Review by carlocaione — APPROVED** [review]

**Pull request merged**